### ISSUE TITLE
Stable box stacking

### ISSUE THREAD

**Issue by Goshido** [Goshido/android-vulkan#19]
It's needed to develop stable box stacking velocity solver. Current implementation is unstable in the following configuration:

**Comment by Goshido** [comment]
Some improvements have been implemented which increased simulation stability:
- penetration slope and restitution slope for velocity solver
- dumping constants have been changed to 0.9
- _Sutherland-Hodgman_ implementation fix
- _Sequential impulse_ implementation fix
- _TBN_ as part of `Contact` struct
- manifold simplification algorithm

The [video](https://drive.google.com/file/d/1vyx8k5loBiBmbW74XthNycLsZHicsJtE/view?usp=sharing) with results.

**Comment by Goshido** [comment]
It was found that it's needed a lot of inverse 4x4 matrix operations if I implement 4 contact manifold simultaneous calculation. There is an article with _SSE_ optimizations. It's [Fast 4x4 Matrix Inverse with SSE SIMD, Explained](https://lxjk.github.io/2017/09/03/Fast-4x4-Matrix-Inverse-with-SSE-SIMD-Explained.html).

After analyzing and porting to [_Neon A64_](https://community.arm.com/developer/ip-products/processors/b/processors-ip-blog/posts/coding-for-neon---part-1-load-and-stores) I able to got result:

```
D/android_vulkan::C++: Results:
    >>>
        Matrices:            40000000 items
        Matrices data:        2.38419 Gb
        Classical:         2234873462 nanoseconds
        Neon:               867129923 nanoseconds
    <<<


D/android_vulkan::C++: Results:
    >>>
        Matrices:            40000000 items
        Matrices data:        2.38419 Gb
        Classical:         2243513539 nanoseconds
        Neon:               866007154 nanoseconds
    <<<


D/android_vulkan::C++: Results:
    >>>
        Matrices:            40000000 items
        Matrices data:        2.38419 Gb
        Classical:         2245197154 nanoseconds
        Neon:               874417000 nanoseconds
    <<<



D/android_vulkan::C++: Results:
    >>>
        Matrices:            40000000 items
        Matrices data:        2.38419 Gb
        Classical:         2214903462 nanoseconds
        Neon:               882053230 nanoseconds
    <<<
```

Long story short: 2.5x boost compare to classical implementation.
